azurerm/internal/services/appplatform/spring_cloud_service_resource.go

```diff
@@ -618,3 +687,25 @@ func flattenArmSpringCloudGitPatternRepository(input *[]appplatform.GitPatternRe
 
 	return results
 }
+
+func flattenArmSpringCloudTrace(input *appplatform.TraceProperties) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	enabled := false
+	appInsightInstrumentationKey := ""
+	if input.Enabled != nil {
+		enabled = *input.Enabled
+	}
+	if input.AppInsightInstrumentationKey != nil {
+		appInsightInstrumentationKey = *input.AppInsightInstrumentationKey
+	}
+
+	return []interface{}{
```

Comment:
if this is Disabled we can return an empty list here:

```suggestion
	if !enabled {
		return []interface{}{}
	}
```